moveit_core/trajectory_processing/include/moveit/trajectory_processing/time_optimal_trajectory_generation.h

```diff
@@ -45,6 +45,15 @@
 
 namespace trajectory_processing
 {
+enum LimitType
```

Comment:
LimitType shouldn't be public if it's only being used by a single private function. Overall, the use of an enum-to-string map just to print a simple log message seems overkill to me. I would be less confusing if you would just simplify the original code with [std::clamp](https://en.cppreference.com/w/cpp/algorithm/clamp)